File: ado/PgWireCoreClient/PgWireAdo/utils/TypesOids.cs


namespace PgWireAdo.utils
{
    public enum TypesOids :int
    {
        Integer = 20,

        Int8 = 20,
        Float8 = 701,
        Int4 = 23,
        Numeric = 1700,
        Float4 = 700,
        Int2 = 21,
        Money = 790,

        // Boolean
        Bool = 16,

        // Geometric
        Box = 603,
        Circle = 718,
        Line = 628,
        LSeg = 601,
        Path = 602,
        Point = 600,
        Polygon = 604,

        // Character
        BPChar = 1042,
        Text = 25,
        Varchar = 1043,
        Name = 19,
        Char = 18,

        // Binary data
        Bytea = 17,

        // Date/Time
        Date = 1082,
        Time = 1083,
        Timestamp = 1114,
        TimestampTz = 1184,
        Interval = 1186,
        TimeTz = 1266,
        Abstime = 702,

        // Network address
        Inet = 869,
        Cidr = 650,
        Macaddr = 829,
        Macaddr8 = 774,

        // Bit string
        Bit = 1560,
        Varbit = 1562,

        // Text search
        TsVector = 3614,
        TsQuery = 3615,
        Regconfig = 3734,

        // UUID
        Uuid = 2950,

        // XML
        Xml = 142,

        // JSON
        Json = 114,
        Jsonb = 3802,
        JsonPath = 4072,

        // public
        Refcursor = 1790,
        Oidvector = 30,
        Int2vector = 22,
        Oid = 26,
        Xid = 28,
        Xid8 = 5069,
        Cid = 29,
        Regtype = 2206,
        Tid = 27,
        PgLsn = 3220,

        // Special
        Record = 2249,
        Void = 2278,
        Unknown = 705,

        // Range types
        Int4Range = 3904,
        Int8Range = 3926,
        NumRange = 3906,
        TsRange = 3908,
        TsTzRange = 3910,
        DateRange = 3912,

        // Multirange types
        Int4Multirange = 4451,
        Int8Multirange = 4536,
        NumMultirange = 4532,
        TsMultirange = 4533,
        TsTzMultirange = 4534,
        DateMultirange = 4535
    }
}

File: pgwire-jdbc-server/pgwire.jdbc.bridge/src/main/java/org/kendar/pgwire/utils/TypesOids.java
package org.kendar.pgwire.utils;

public class TypesOids {
    public static int Int8         = 20;
    public static int Float8       = 701;
    public static int Int4         = 23;
    public static int Numeric      = 1700;
    public static int Float4       = 700;
    public static int Int2         = 21;
    public static int Money        = 790;

    // Boolean
    public static int Bool        = 16;

    // Geometric
    public static int Box         = 603;
    public static int Circle      = 718;
    public static int Line        = 628;
    public static int LSeg        = 601;
    public static int Path        = 602;
    public static int Point       = 600;
    public static int Polygon     = 604;

    // Character
    public static int BPChar      = 1042;
    public static int Text        = 25;
    public static int Varchar     = 1043;
    public static int Name        = 19;
    public static int Char        = 18;

    // Binary data
    public static int Bytea       = 17;

    // Date/Time
    public static int Date        = 1082;
    public static int Time        = 1083;
    public static int Timestamp   = 1114;
    public static int TimestampTz = 1184;
    public static int Interval    = 1186;
    public static int TimeTz      = 1266;
    public static int Abstime     = 702;

    // Network address
    public static int Inet        = 869;
    public static int Cidr        = 650;
    public static int Macaddr     = 829;
    public static int Macaddr8    = 774;

    // Bit string
    public static int Bit         = 1560;
    public static int Varbit      = 1562;

    // Text search
    public static int TsVector    = 3614;
    public static int TsQuery     = 3615;
    public static int Regconfig   = 3734;

    // UUID
    public static int Uuid        = 2950;

    // XML
    public static int Xml         = 142;

    // JSON
    public static int Json        = 114;
    public static int Jsonb       = 3802;
    public static int JsonPath    = 4072;

    // public
    public static int Refcursor   = 1790;
    public static int Oidvector   = 30;
    public static int Int2vector  = 22;
    public static int Oid         = 26;
    public static int Xid         = 28;
    public static int Xid8        = 5069;
    public static int Cid         = 29;
    public static int Regtype     = 2206;
    public static int Tid         = 27;
    public static int PgLsn       = 3220;

    // Special
    public static int Record      = 2249;
    public static int Void        = 2278;
    public static int Unknown     = 705;

    // Range types
    public static int Int4Range   = 3904;
    public static int Int8Range   = 3926;
    public static int NumRange    = 3906;
    public static int TsRange     = 3908;
    public static int TsTzRange   = 3910;
    public static int DateRange   = 3912;

    // Multirange types
    public static int Int4Multirange   = 4451;
    public static int Int8Multirange   = 4536;
    public static int NumMultirange    = 4532;
    public static int TsMultirange     = 4533;
    public static int TsTzMultirange   = 4534;
    public static int DateMultirange   = 4535;
}
